libs/langchain-google-common/src/media.ts

```diff
@@ -0,0 +1,775 @@
+      GoogleResponse,
+      AuthOptions
+    >(params, this.caller, this.client);
+  }
+}
+
+export type AIStudioFileState = "PROCESSING" | "ACTIVE" | "FAILED" | "STATE_UNSPECIFIED";
+
+export type AIStudioFileVideoMetadata = {
+  videoMetadata: {
+    videoDuration: string,   // Duration in seconds, possibly with fractional, ending in "s"
+  }
+}
+
+export type AIStudioFileMetadata = AIStudioFileVideoMetadata;
+
+export interface AIStudioFileObject {
+  name?: string,
+  displayName?: string,
+  mimeType?: string,
+  sizeBytes?: string,      // int64 format
+  createTime?: string,     // timestamp format
+  updateTime?: string,     // timestamp format
+  expirationTime?: string, // timestamp format
+  sha256Hash?: string,     // base64 encoded
+  uri?: string,
+  state?: AIStudioFileState,
+  error?: {
+    code: number,
+    message: string,
+    details: Record<string,unknown>[],
+  },
+  metadata?: AIStudioFileMetadata,
+}
+
+export class AIStudioMediaBlob extends MediaBlob {
+
+  _valueAsDate(value: string): Date {
+    if (!value) {
+      return new Date(0);
+    }
+    return new Date(value);
+  }
+
+  _metadataFieldAsDate(field: string): Date {
+    return this._valueAsDate(this.metadata?.[field]);
+  }
+
+  get createDate(): Date {
+    return this._metadataFieldAsDate("createTime");
+  }
+
+  get updateDate(): Date {
+    return this._metadataFieldAsDate("updateTime");
+  }
+
+  get expirationDate(): Date {
+    return this._metadataFieldAsDate("expirationTime");
+  }
+
+  get isExpired(): boolean {
+    const now = (new Date()).toISOString();
+    const exp = this.metadata?.expirationTime ?? now;
+    return exp <= now;
+  }
+}
+
+export interface AIStudioFileGetResponse extends GoogleResponse {
+  data: AIStudioFileObject;
+}
+
+export interface AIStudioFileSaveResponse extends GoogleResponse {
+  data: {
+    file: AIStudioFileObject,
+  }
+}
+
+export interface AIStudioFileListResponse extends GoogleResponse {
+  data: {
+    files: AIStudioFileObject[],
+    nextPageToken: string,
+  }
+}
+
+export type AIStudioFileResponse =
+  | AIStudioFileGetResponse
+  | AIStudioFileSaveResponse
+  | AIStudioFileListResponse;
+
+export interface AIStudioFileConnectionParams {}
+
+export interface AIStudioFileUploadConnectionParams<AuthOptions>
+  extends GoogleUploadConnectionParams<AuthOptions>,
+    AIStudioFileConnectionParams {}
+
+export class AIStudioFileUploadConnection<
+  AuthOptions
+> extends GoogleMultipartUploadConnection<
+  AsyncCallerCallOptions,
+  AIStudioFileSaveResponse,
+  AuthOptions
+> {
+
+  apiVersion = "v1beta";
+
+  async buildUrl(): Promise<string> {
+    return `https://generativelanguage.googleapis.com/upload/${this.apiVersion}/files`;
+  }
+}
+
+export interface AIStudioFileDownloadConnectionParams<AuthOptions>
+  extends AIStudioFileConnectionParams,
+    GoogleConnectionParams<AuthOptions> {
+  method: GoogleAbstractedClientOpsMethod;
+  name: string;
+}
+
+export class AIStudioFileDownloadConnection<
```

Comment:
Could we rely on an official SDK for some of this or are there the same problems?

Reply:
Relying on the official SDK means importing another package, and I'm not sure what it gets us.